include/fi_mem.h

```diff
@@ -62,15 +62,69 @@ static inline void *mem_dup(const void *src, size_t size)
 	return dest;
 }
 
+
+/*
+ * Buffer pool (free stack) template
+ */
+#define FREESTACK_EMPTY	-1
+
+#define freestack_isempty(fs)	((fs)->next == FREESTACK_EMPTY)
+#define freestack_push(fs, x)					\
+{								\
+	*(int *) &(fs)->buf[x] = (fs)->next;			\
```

Comment:
Why casting to an `int *` here? 

Shouldn't this be part of the freestack template and be casting to `entrytype`?